src/ServiceClient/Implements/SocketDataTransfer.cs

```diff
@@ -0,0 +1,34 @@
+﻿using ServiceClient.Abstractions;
+using System.Net.WebSockets;
+using System.Text;
+
+namespace ServiceClient.Implements
+{
+    public class SocketDataTransfer : ISocketDataTransfer
+    {
+        private const int READ_BUFFER_SIZE = 4096;
+
+        public async Task<string> ReceiveAsync(ClientWebSocket socket, CancellationToken cancellationToken)
+        {
+            StringBuilder sb = new StringBuilder();
+            var buffer = new byte[READ_BUFFER_SIZE];
+            WebSocketReceiveResult response;
+            do
+            {
+                response = await socket.ReceiveAsync(buffer, CancellationToken.None);
```

Comment:
I think the response should be validated, otherwise if there is a problem receiving  you may get and exception